src/components/templates/TalkShowStudio/TalkShowStudio.tsx

```diff
@@ -94,56 +94,56 @@ export const TalkShowStudio: FC<TalkShowStudioProps> = ({ venue }) => {
       ),
     [userOnStageSharingScreen?.data, remoteUsers, venue.id]
   );
-  const remoteCameraTrack = useMemo(
-    () =>
-      remoteUsers.find(
-        ({ uid }) =>
-          `${uid}` ===
-          userOnStageSharingScreen?.data?.[`${venue.id}`]?.cameraClientUid
-      ),
-    [userOnStageSharingScreen?.data, remoteUsers, venue.id]
-  );
 
   const remoteUsersPlayers = useMemo(() => {
+    const uniqueUsersIds = new Set();
     const setRemoteUserAvatar = (remoteUserId: number | string) => {
       if (!venue.id) return;
-      const remoteUser = stage.peopleOnStage.find(
+
+      return stage.peopleOnStage.find(
         ({ data }) =>
           data?.[`${venue.id}`]?.cameraClientUid === `${remoteUserId}`
       );
-      return remoteUser;
     };
-    return remoteUsers
-      .filter(
-        ({ uid }) =>
-          `${uid}` !==
-          userOnStageSharingScreen?.data?.[`${venue.id}`]?.cameraClientUid
-      )
-      .filter(
-        ({ uid }) =>
-          `${uid}` !==
-          userOnStageSharingScreen?.data?.[`${venue.id}`]?.screenClientUid
-      )
-      .map(
-        (user) =>
-          user.uid !== screenClient.uid &&
-          user.uid !== cameraClient.uid && (
-            <div key={user.uid}>
-              {user.hasVideo && (
-                <Player
-                  showButtons
-                  user={setRemoteUserAvatar(user.uid)}
-                  videoTrack={user.videoTrack}
-                  audioTrack={user.audioTrack}
-                  isCamOn={user.hasVideo}
-                  isMicOn={user.hasAudio}
-                  containerClass="TalkShowStudio__mode--play"
-                />
-              )}
-            </div>
-          )
+
+    return remoteUsers.map((agoraRemoteUser) => {
+      const user = setRemoteUserAvatar(agoraRemoteUser.uid);
+      if (!user) return null;
+
+      const isUniqueUser = !uniqueUsersIds.has(user.id);
+      if (isUniqueUser) {
+        uniqueUsersIds.add(user.id);
+      }
+
+      return (
+        agoraRemoteUser.uid !== screenClient.uid &&
+        agoraRemoteUser.uid !== cameraClient.uid && (
+          <div key={agoraRemoteUser.uid}>
+            {isUniqueUser && (
+              <Player
+                showButtons
+                user={setRemoteUserAvatar(agoraRemoteUser.uid)}
+                videoTrack={agoraRemoteUser.videoTrack}
+                audioTrack={agoraRemoteUser.audioTrack}
+                isCamOn={agoraRemoteUser.hasVideo}
+                isMicOn={agoraRemoteUser.hasAudio}
+                toggleCam={isUserOwner ? stage.toggleUserCamera : undefined}
+                toggleMic={isUserOwner ? stage.toggleUserMicrophone : undefined}
+                containerClass="TalkShowStudio__mode--play"
+              />
```

Comment:
There are a LOT of props needing to be passed through to this component.. I expect there should probably be a better pattern to use here.

Reply:
Duplicate of changes made in https://github.com/sparkletown/sparkle/pull/1653/files#r659514704